src/smart-components/role/add-role-permissions/add-role-permission-view.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
+import { listPermissions, listPermissionOptions } from '../../../redux/actions/permission-action';
+import { TableToolbarView } from '../../../presentational-components/shared/table-toolbar-view';
+
+const columns = ['Application', 'Resource type', 'Operation'];
+const selector = ({
+  permissionReducer: {
+    permission,
+    isLoading,
+    options: { application, operation, resource },
+  },
+  roleReducer: { isRecordLoading, selectedRole },
+}) => ({
+  permissions: permission.data.map(({ application, resource_type: resource, verb, permission } = {}) => ({
+    application,
+    resource,
+    operation: verb,
+    uuid: permission,
+  })),
+  pagination: permission.meta,
+  isLoading: isLoading || isRecordLoading,
+  baseRole: selectedRole,
+  applicationOptions: application.data,
+  resourceOptions: resource.data,
+  operationOptions: operation.data,
+});
+
+const AddRolePermissionView = ({ selectedPermissions, setSelectedRolePermissions }) => {
+  const dispatch = useDispatch();
+  const { permissions, isLoading, pagination } = useSelector(selector, shallowEqual);
+  const fetchData = (apiProps) => dispatch(listPermissions(apiProps));
+  const fetchOptions = (apiProps) => dispatch(listPermissionOptions(apiProps));
+
+  const createRows = (permissions, expanded, checkedRows = []) =>
+    permissions.map(({ application, resource, operation, uuid }) => ({
+      uuid: `${application}:${resource}:${operation}`,
+      cells: [application, resource, operation],
+      selected: Boolean(checkedRows && checkedRows.find((row) => row.uuid === uuid)),
```

Comment:
This can be simpliied with [`some` function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)
```suggestion
      selected: checkedRows?.some((row) => row.uuid === uuid),
```